### PR TITLE
[STRMCMP-659] Add watermark support to Flink Kafka consumer

### PR DESCRIPTION
**Pull request by tweise**

### CODE REVIEW

File: runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java [hunk 1]
```diff
@@ -102,32 +102,41 @@ private void translateKafkaInput(
...
+    Preconditions.checkNotNull(consumerProps, "'properties' need to be set");
+    properties.putAll(consumerProps);
+
     logger.info("Kafka consumer for topic {} with properties {}", topic, properties);
 
-    DataStreamSource<WindowedValue<byte[]>> source =
+    FlinkKafkaConsumer011<WindowedValue<byte[]>> kafkaSource =
+        new FlinkKafkaConsumer011<>(topic, new ByteArrayWindowedValueSchema(), properties);
+
+    kafkaSource.setStartFromLatest();
```
Comment (JdevEclipse) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java:133]: Do you want to integrate the offset fix here if it is not too much work?

**Comment by JdevEclipse** [comment]
👍

File: runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java (hunk 1 above)
Comment (tweise) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java:133]: That's a separate change.

**Comment by tweise** [comment]
This isn't ready yet, I need to do testing.

**Pull request merged**